### PR TITLE
BUILD-348: share discoverability ('use' verb, namespace scope roles); inject build-284 in repo

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @adambkaplan 
/assign @coreydaley 
@deads2k FYI (if anything else, for api godoc)

So this PR proves out our recent discussions in slack and @coreydaley 's https://github.com/openshift/api/pull/979

Highlights:
- switches to only a 'use' verb when seeing if the Pod SA can use / mount a SharedResource
- I replaced use of ClusterRole/ClusterRoleBinding in tests/examples to vet out the discussion point around allowing namespace admins to grant permissions to SA's in their namespaces for shares they can discover based on separate roles etc. around getting and listing SharedResources
- I pulled in @coreydaley 's PR change into this repo's current local API def, including updating the godoc based on the simplification @deads2k noted because of the switch to the 'use' verb , and updated all the code/tests with the changes from that
- presumably when @coreydaley 's PR merges his openshift/api vendor into this repo should be greatly simplified
- I've deferred on the naming discussion @adambkaplan is having with Jan on the groupname ... left it as is from @coreydaley 's PR for now

I'm still going to put up a openshift/enhancements update, but I am currently of the opinion this PR progresses us on the path, could be merged after review, and then follow up PRs can come if the EP merge or api merge results in addition adjustments.

### CODE REVIEW

**Comment by gabemontero** [comment]
all green e2e's @adambkaplan @coreydaley

**Review by adambkaplan — COMMENTED** [review]
/approve

Nits, but otherwise looks good

File: pkg/api/sharedresource/v1alpha1/share_type.go [hunk 1]
```diff
@@ -13,58 +8,120 @@ import (
...
+//
+// spec:
+//  volumes:
+//  - name: shared-secret
+//    csi:
+//      driver: sharedresource.storage.openshift.io
+//      volumeAttributes:
+//        sharedResource: my-share
+//
+// For the mount to be successful, the pod's service account must be granted permission to 'use' the named SharedResource object
```
Comment (gabemontero) [inline pkg/api/sharedresource/v1alpha1/share_type.go:23]: so starting here @coreydaley was my attempt to take what @deads2k and I discussed in slack in reference to your openshift/api PR and apply "corrections" in the godoc

File: pkg/api/sharedresource/v1alpha1/share_type.go [hunk 2]
```diff
@@ -13,58 +8,120 @@ import (
...
+// within its namespace with an appropriate Role and RoleBinding.  For compactness, here are example `oc` invocations for creating
+// such Role and RoleBinding objects.
+//
+//  `oc create role shared-resource-my-share --verb=use --resource=sharedresources.storage.openshift.io --resource-name=my-share`
+//  `oc create rolebinding shared-resource-my-share --role=shared-resource-my-share --serviceaccount=my-namespace:default`
+//
+// Administrators can create separate Roles and RoleBindings for their users to be able the list and/or view the
+// available cluster scoped `SharedResources` objects.
```
Comment (gabemontero) [inline pkg/api/sharedresource/v1alpha1/share_type.go:31]: this is the end of where I tried to update godoc @coreydaley in reference to discussion with @deads2k in slack

ideally you can replicate this in your openshift/api PR, and then, with the resolution of the group name that @adambkaplan will be striving for with @jsafrane and @deads2k your openshift/api PR will be "unblocked", or at least at the point where we are just iterating on grammar and word choice vs. major concepts and architectural viability

File: docs/simple-example.md [hunk 3]
```diff
@@ -19,40 +19,43 @@ metadata:
...
-  name: projected-resource-my-share
+  name: shared-resource-my-share
+  namespace: my-csi-app-namespace
 rules:
   - apiGroups:
-      - projectedresource.storage.openshift.io
+      - storage.openshift.io
     resources:
-      - shares
+      - sharedresourcess
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply docs/simple-example.md:30]: yep ... fortunately it is correct in the actual yaml files in the examples subdir ;-)

**Comment by gabemontero** [comment]
getting 500s from CIs registry on startup of all jobs

will sit tight and retry tests later

**Pull request merged**